### DISCUSSION TITLE
OpenAirInterface5G with AntSDR Support

### DISCUSSION THREAD

**Discussion by RabihND** [MicroPhase/antsdr_uhd#85] [Show and tell]
Hello AntSDR community,

I wanted to share a development related to AntSDR integration. I've recently created a forked version of the OpenAirInterface5G project with added support for AntSDR U220, allowing me to use it in my projects. 
To view the changes I've made, please check the commits in my repository.
Here are the details:

1. Repository Name: **OAI5G_AntSDR**
2. GitHub Link: https://github.com/RabihND/OAI5G_AntSDR.git

Key Features:
- Forked from the official OpenAirInterface5G repository at `69a84825fb52cfa0bbd728e0ba137ae4e72f695a` (September 4, 2024)
- Added support for AntSDR hardware
- Provides Docker files for building the gNB

Purpose:
This project aims to bridge the gap between OpenAirInterface5G and AntSDR, allowing for easier integration and deployment of 5G networks using AntSDR hardware.

How it works:
- The repository contains the standard OpenAirInterface5G structure
- Additional files and configurations have been added to support AntSDR
- Docker files are provided to simplify the build process for the gNB

Why it matters:
- Enables researchers and developers to leverage AntSDR hardware with OpenAirInterface5G

I'd love to hear your feedback and encourage collaboration. Feel free to clone the repository, test it out, and contribute improvements!

Best regards,

**Comment by i-demetriou** [comment]
Hi @RabihND 

Thanks for this. We also want to try to get our AntSDR working with open air interface.

From what I see, your diff is concentrating on patching the `build_oai` and the Dockerfiles to build the UHD driver of this repo.

I tried to run it, but I get a lot of error messages from the `[PHY]` about samples.

```log
[PHY]   rx_rf: Asked for 11520 samples, got 6092 from USRP
[PHY]   problem receiving samples
LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL
LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL
LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL
LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL
LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL
LLLLLLLLLLLLLLLLLLD[HW]   [recv] received 2864 samples out of 11520
L[HW]   Time: 7.92608 s
ERROR_CODE_OVERFLOW (Out of sequence error)
```

Maybe there is something wrong with the config I provided?

**Reply by RabihND** [reply]
[GitHub Issue #2](https://github.com/RabihND/OAI5G_AntSDR/issues/2)
